pkg/dependencies/factory.go

```diff
@@ -579,39 +578,19 @@ func (f *Factory) WithTroubleshoot() *Factory {
 	return f
 }
 
-func (f *Factory) WithHelmSecure() *Factory {
+func (f *Factory) WithHelm(insecure bool) *Factory {
 	f.WithExecutableBuilder()
 
 	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
-		if f.dependencies.HelmSecure != nil {
+		if f.dependencies.Helm != nil {
 			return nil
 		}
```

Comment:
The insecure flag here won't work because once the helm executable is built, it is not going to build it again

Reply:
What do you mean by this? It gets built on every command execution?

Reply:
There was an if condition which would prevent the executable from being re-built if wasn't nil. I think this is resolved in the latest commit